test/lib/diff.js

```diff
@@ -0,0 +1,1095 @@
+    })
+  })
+
+  t.test('cwd not a project dir', t => {
+    t.plan(2)
+
+    const path = t.testdir({})
+    rlp = () => {
+      throw new Error('ERR')
+    }
+    libnpmdiff = async ([a, b], opts) => {
+      t.equal(a, 'bar@latest', 'should use latest tag')
+      t.equal(b, 'bar-fork@latest', 'should use latest tag')
+    }
+
+    npm.flatOptions.diff = ['bar', 'bar-fork']
+    npm.flatOptions.prefix = path
+
+    diff([], err => {
+      if (err)
+        throw err
+    })
+  })
+
+  t.end()
+})
+
+t.test('various options', t => {
+  t.test('using --name-only option', t => {
+    t.plan(1)
+
+    npm.flatOptions.diffNameOnly = true
+
+    libnpmdiff = async ([a, b], opts) => {
+      t.match(opts, {
+        ...npm.flatOptions,
+        diffNameOnly: true,
+      }, 'should forward nameOnly=true option')
+    }
+
+    diff([], err => {
+      if (err)
+        throw err
+    })
+  })
+
+  t.test('set files after both versions', t => {
+    t.plan(3)
+
+    npm.flatOptions.diff = ['2.1.4', '3.0.0']
+
+    libnpmdiff = async ([a, b], opts) => {
+      t.equal(a, 'foo@2.1.4', 'should use expected spec')
+      t.equal(b, 'foo@3.0.0', 'should use expected spec')
+      t.match(opts, {
+        ...npm.flatOptions,
+        diffFiles: [
+          './foo.js',
+          './bar.js',
+        ],
+      }, 'should forward diffFiles values')
+    }
+
+    diff(['./foo.js', './bar.js'], err => {
+      if (err)
+        throw err
+    })
+  })
+
+  t.test('set files no diff args', t => {
+    t.plan(3)
+
+    const path = t.testdir({})
+    rlp = async () => 'my-project'
+    libnpmdiff = async ([a, b], opts) => {
+      t.equal(a, 'my-project@latest', 'should have default spec')
+      t.equal(b, `file:${path}`, 'should compare to cwd')
+      t.match(opts, {
+        ...npm.flatOptions,
+        diffFiles: [
+          './foo.js',
+          './bar.js',
+        ],
+      }, 'should forward all remaining items as filenames')
+    }
+
+    npm.flatOptions.prefix = path
+    diff(['./foo.js', './bar.js'], err => {
+      if (err)
+        throw err
+    })
+  })
+
+  t.test('using diff option', t => {
+    t.plan(1)
+
+    npm.flatOptions.diffContext = 5
+    npm.flatOptions.diffIgnoreWhitespace = true
+    npm.flatOptions.diffNoPrefix = false
+    npm.flatOptions.diffSrcPrefix = 'foo/'
+    npm.flatOptions.diffDstPrefix = 'bar/'
+    npm.flatOptions.diffText = true
+
+    libnpmdiff = async ([a, b], opts) => {
+      t.match(opts, {
+        ...npm.flatOptions,
+        diffContext: 5,
+        diffIgnoreWhitespace: true,
+        diffNoPrefix: false,
+        diffSrcPrefix: 'foo/',
+        diffDstPrefix: 'bar/',
+        diffText: true,
+      }, 'should forward diff options')
+    }
+
+    diff([], err => {
+      if (err)
+        throw err
+    })
+  })
+
+  t.end()
+})
+
+t.test('too many args', t => {
+  npm.flatOptions.diff = ['a', 'b', 'c']
+  diff([], err => {
+    t.match(
+      err,
+      /Can't use more than two --diff arguments./,
```

Comment:
Would be nice if nopt had a way to specify the max length for array configs and not have to do this ad hoc.  Yet another reason to switch to jackspeak ;P